File: gcd_recursive.js
#!/usr/bin/env node
// http://rosettacode.org/wiki/Greatest_common_divisor

function gcdRecursive(a, b) {
  return b == 0 ? a : gcdRecursive(b, (a % b))
}

console.log(gcdRecursive(36, 33))
console.log(gcdRecursive(36, 16))
console.log(gcdRecursive(8, 24))

File: gcd_recursive.py
#!/usr/bin/env python

def greatest_common_divisor(x,y):
  if y == 0:
    return x
  else:
    return greatest_common_divisor(y,(x%y))

for pair in [(36,33),(36,16),(8,24)]:
  print greatest_common_divisor(pair[0],pair[1])